SEED-platform/seed
# meter usage by month down to the second

#### Any background context you want to provide?
Meter usage is uploaded on a per building bases. Based on ESPM meter usage data is in one-month chunks, however, it is often uploaded across a month (April 15th to May 15th). Currently, SEED can handle sorting meter data when the start and end dates are within a single month.

Monthly meter usages should be aggregated based on partial monthly values as explained by an excerpt from Portfolio Manager Energy Star, page 6, section 2: https://portfoliomanager.energystar.gov/pdf/reference/Climate%20and%20Weather.pdf
<img width="655" alt="Screen Shot 2022-04-21 at 4 06 24 PM" src="https://user-images.githubusercontent.com/58446472/164559525-287f5a00-b9d9-482d-9f8d-ac2c0aaeadff.png">


#### What's this PR do?
Allows SEED to sort meter usage when readings span a month. The logic finds the total number of seconds within the date ranges. Then using a linear relationship determines the reading that falls within month boundaries and adds that calculated reading to the corresponding monthly total. 

#### How should this be manually tested?
Import the property AND meter data for 
[ESPM_JCC_Custom_Download_BETTERTest.xlsx](https://github.com/SEED-platform/seed/files/8557649/ESPM_JCC_Custom_Download_BETTERTest.xlsx)


On Develop: 
view property `PM Property ID: 7815736` > meters > notice date periods are contained to a single month > interval > month > successful
view property `PM Property ID: 3207429` > meters > notice date periods are not contained to a single month > interval > month > No Data

On this branch:
view property `PM Property ID: 7815736` > meters > notice date periods are contained to a single month > interval > month > successful (ensure same readings as on develop branch)
view property `PM Property ID: 3207429` > meters > notice date periods are not contained to a single month > interval > month > successful

#### What are the relevant tickets?
#2685

#### Screenshots (if appropriate)
